classes/search/SearchFileParser.php

```diff
@@ -127,22 +131,23 @@ public static function fromFile($submissionFile)
      * @param string $type
      * @param string $path
      *
-     * @return SearchFileParser
+     * @return ?SearchFileParser
      */
     public static function fromFileType($type, $path)
     {
-        switch ($type) {
-            case 'text/plain':
-                return new self($path);
-                break;
-            case 'text/html':
-            case 'text/xml':
-            case 'application/xhtml':
-            case 'application/xml':
-                return new \PKP\search\SearchHTMLParser($path);
-                break;
-        }
-        return new \PKP\search\SearchHelperParser($type, $path);
+        $type = Config::getVar('search', "index[{$type}]")
```

Comment:
This kind of double-ternary construct is pretty hard to read. I guess you could simplify it by giving a default value to `Config::getVar`?
```php
$type = Config::getVar('search', "index[{$type}]", 'default');
```
OTOH this will give you `default` if the configuration line is not specified at all, but will still respect an `Off` setting if you've configured that indexer to be entirely disabled.

Reply:
I'm ok with the nested stuff, as long as they are not very long, but no worries, I'll update 😁
Yeah, I saw the `getVar()` accepts a default, but it's not suitable for my use case.